src/bson_value.ts

```diff
@@ -2,10 +2,33 @@ import { BSON_MAJOR_VERSION } from './constants';
 import { type InspectFn } from './parser/utils';
 import { BSON_VERSION_SYMBOL } from './constants';
 
+/** @public */
+export type BSONTypeTag =
+  | 'BSONRegExp'
+  | 'BSONSymbol'
+  | 'ObjectId'
+  | 'Binary'
+  | 'Decimal128'
+  | 'Double'
+  | 'Int32'
+  | 'Long'
+  | 'MaxKey'
+  | 'MinKey'
+  | 'Timestamp'
+  | 'Code'
+  | 'DBRef';
```

Comment:
Not strictly necessary, but I do think a proper enumeration of types would generally be helpful to downstream consumers (I can't think of a situation in which this would not be helpful right now, at least).